### PR TITLE
Update docs to an actual state as for 0.7.0

### PR DESCRIPTION
**Pull request by vkostyukov**
WIP: Pushing the branch to make sure it renders correctly.

### CODE REVIEW

**Comment by vkostyukov** [comment]
Docs are in progress. I've finished `request.md` and `response.md`. `route.md` is on my list of todos.

**Comment by vkostyukov** [comment]
This is ready to be reviewed.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -12,27 +12,30 @@ elsewhere.
...
 
 That said, the Scala style checker `sbt scalastyle` should pass on the code. 
 
 ## Write Tests
-Finch uses [ScalaTest][2] with the following settings:
+Finch uses both [ScalaTest][2] and [ScalaCheck][3] with the following settings:
 
-* Every test should be a `FlatSpec` with `Matchers` mixed in
+* Every test should be a `FlatSpec` with `Matchers` and `Checkers` mixed in
 * An assertion should be done with `x shouldBe y`
```
Comment (travisbrown) [inline CONTRIBUTING.md]: `check` expects its argument to return a boolean (or something similar—not `Unit`, in any case), which means that assertions in properties have to be written with `===`.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -12,27 +12,30 @@ elsewhere.
...
 * Exceptions should be intercepted with `an [Exception] shouldBe thrownBy(x)`
 
 ## Write Docs
-Write clean and simple docs in the `docs.md` file.
+Write clean and simple docs in the `docs` folder.
 
 ## Submit a PR
 * PR should be submitted from a separate branch (use `git checkout -b "fix-123"`)
-* PR should contain only one commit (use `git commit --amend` and `git --force push`)
+* PR should contain only one commit (use `git commit --amend` and `git --force push` or [squash][4] existing commits into one)
```
Comment (travisbrown) [inline CONTRIBUTING.md]: Maybe "generally"? I think there are cases where keeping multiple commits is useful. I could go either way, though.

Comment (vkostyukov) [inline CONTRIBUTING.md]: Yeah. There are exceptions though. I will add "generally".

**Comment by vkostyukov** [comment]
Thanks @travisbrown! I've updated the PR.

File: docs/route.md [hunk 3]
```diff
@@ -142,8 +148,49 @@ endpoint(request) handle {
 }
 ```
 
+### Coproduct Routers
+
+The `|` compositor is pretty useful for the simple cases when both underlying types `A` and `B` may be substituted with
+a single super type `C` such that `C >: A` and `C >: B` (usually that means that `A =:= B`). Although, it should be also
+possible to compose two routers of completely different types. This is pretty duable the `:+:` compositor that composes
```
Comment (travisbrown) [inline docs/route.md]: s/duable/doable with/

File: docs/route.md [hunk 4]
```diff
@@ -142,8 +148,49 @@ endpoint(request) handle {
...
+two routers into a `Router[C <: Coproduct]`, where `Coproduct` [Shapeless' disjoint union type][2].
```
Comment (travisbrown) [inline docs/route.md]: "`Coproduct` is"

**Comment by travisbrown** [comment]
This looks generally good to me!—I'm happy to make a closer pass over the rest, but probably won't be able to get to it until tomorrow evening.

**Comment by vkostyukov** [comment]
I decided to go with 12 chars limit since IDEA doesn't support ([yet](https://youtrack.jetbrains.com/issue/IDEA-59662)) setting the customized limit for Markdown-files. Although, I would be happy to reduce the limit later.

**Comment by vkostyukov** [comment]
Let's merge this then!

**Pull request merged**